.github/workflows/pipeline.yml

```diff
@@ -5,21 +5,43 @@ on:
     branches:
       - main
   workflow_dispatch: # Can be triggered manually from a branch
-
+    inputs:
+      environment:
+        description: 'Deployment Environment (valid values: "development", "production")'
+        required: true
+        default: 'development'
+      version:
+        description: 'Application version to deploy'
+        required: true
+        
 jobs:
   build:
     name: Build
     uses: ./.github/workflows/build.yml
     with:
       push: true
     secrets: inherit
+  
 
-  deploy-to-dev:
+  deploy_to_dev:
     name: Deploy to dev
     uses: ./.github/workflows/deploy.yml
     needs: build
     with:
       github_environment: development
       environment: dev
       version: ${{ needs.build.outputs.version }}
+    secrets: inherit
+
+
+  deploy_to_prod:
+    name: Deploy to prod
+    uses: ./.github/workflows/deploy.yml
+    needs:
+      - deploy_to_dev # wait for the deploy_to_dev job to complete
+    with:
+      github_environment: production
+      environment: prod
+      version: ${{ github.event.inputs.version }}
+    if: github.event_name != 'schedule' # Skip this job if the workflow is triggered by schedule
```

Comment:
Are we ever going to run this on a schedule?

Reply:
Removing now, however can be used if at all we find there is a need for regular updates, scheduled-maintenance etc